mitosheet/mitosheet/step_performers/graph_steps/graph.py

```diff
@@ -102,8 +107,21 @@ def execute(cls, prev_state: State, params: Dict[str, Any]) -> Tuple[State, Opti
         y_axis_column_ids = graph_creation["y_axis_column_ids"] if graph_creation["y_axis_column_ids"] is not None else []
         y_axis_column_headers = prev_state.column_ids.get_column_headers_by_ids(sheet_index, y_axis_column_ids)
 
-        # Validate optional parameters 
-        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() else None
+        # Validate optional parameters that are available for all graph types
+        color_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["color"]) if 'color' in graph_creation.keys() and graph_creation["color"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_col_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["facet_col_column_id"]) if 'facet_col_column_id' in graph_creation.keys() and graph_creation["facet_col_column_id"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_row_column_header = prev_state.column_ids.get_column_header_by_id(sheet_index, graph_creation["facet_row_column_id"]) if 'facet_row_column_id' in graph_creation.keys() and graph_creation["facet_row_column_id"] in prev_state.column_ids.column_id_to_column_header[sheet_index].keys() else None
+        facet_col_wrap = int(graph_creation['facet_col_wrap']) if 'facet_col_wrap' in graph_creation.keys() else None
+        facet_col_spacing = float(graph_creation['facet_col_spacing']) if 'facet_col_spacing' in graph_creation.keys() else None
+        facet_row_spacing = float(graph_creation['facet_row_spacing']) if 'facet_row_spacing' in graph_creation.keys() else None
+
+        # Validate parameters that are only available for some graph types
+        # Note: We trust the parameters from the frontend, so we don't make sure the params fit the graph type here
+        points = graph_creation['points'] if 'points' in graph_creation.keys() else None
+        line_shape = graph_creation['line_shape'] if 'line_shape' in graph_creation.keys() else None
+        histnorm = graph_creation['histnorm'] if 'histnorm' in graph_creation.keys() else None
+        histfunc = graph_creation['histfunc'] if 'histfunc' in graph_creation.keys() else None
+        nbins = int(graph_creation['nbins']) if 'nbins' in graph_creation.keys() else None
```

Comment:
For the ones that don't need to be cast, etc. we could probably just pass them through as params, so we don't need to do all of this validation, because we end up just checking if it is not None, then add it. 

If we just don't do this check, then the param won't be there initially for us to check.